### ISSUE TITLE
Add a custom transport API

### ISSUE THREAD

**Issue by bymaximus** [minio/minio-js#523]
No way to set 'timeout' for request like 'statObject'?

**Comment by harshavardhana** [comment]
> No way to set 'timeout' for request like 'statObject'?

There is no way. is there a specific reason?

**Comment by bymaximus** [comment]
Yes, at least for the project im working, for example I need check if a file/object exists at minio storage server which continuous receive tons of operations and maybe can be busy or even offline for some reason and make current api call fail.

Also, for example if you try to 'statObject' at any server not running minio it will take aprox 2min (120sec) to call callback function.
```
start time Mon Dec 12 2016 21:24:22 GMT-0200 (BRST)
inside callback, end time 127.385 Mon Dec 12 2016 21:26:29 GMT-0200 (BRST)
{ [Error: connect ETIMEDOUT 142.54.164.14:9000]
  code: 'ETIMEDOUT',
  errno: 'ETIMEDOUT',
  syscall: 'connect',
  address: '142.54.164.14',
  port: 9000 }
```

I think you can return the 'request' object as api function return value to enable us to abort the request if needed instead of add entire timeout code, this way we can abort api call request when needed

**Comment by harshavardhana** [comment]
This is perhaps fixed by - bbbebe595d128d1dff3fdf6e357f430ac0885d12 and perhaps manages connections properly. Have you tried with the latest release @bymaximus ?

**Comment by krishnasrinivas** [comment]
@bymaximus since JS is not strict, you can do this:

minioClient = new Minio.Client({....})
minioClient.transport = customTransport

But we don't recommend this as your application can break in case we change things internally.
